src/openscm_units/_unit_registry.py

```diff
@@ -312,6 +312,32 @@ class ScmUnitRegistry(pint.UnitRegistry):
 
     _contexts_loaded = False
 
+    def __init__(self, *args, metric_conversions_csv=None, **kwargs):
+        """
+        Initialise the unit registry
+
+        Parameters
+        ----------
+        metric_conversions_csv : str
+            csv file containing the metric conversions. If not supplied,
+            the internal metric conversions csv is used.
+
+        *args
+            Passed to the ``__init__`` method of the super class
+
+        **kwargs
+            Passed to the ``__init__`` method of the super class
+        """
+        if metric_conversions_csv is None:
+            metric_conversions_csv = path.join(
+                path.dirname(path.abspath(__file__)),
+                "data",
+                "metric_conversions.csv",
+            )
```

Comment:
While it is not necessary, I think best practice would be to use [importlib](https://importlib-resources.readthedocs.io/en/latest/using.html) here. The advantage is that it also works within a zipped python module and when compiling a python module using the various tools that exist etc. The common pattern would be:

```python
from . import data
fd = importlib.resources.open_text(data, "metric_conversions.csv")
```

However, I'm not totally sure how that would be combined with the lazy loading. Maybe just save the `None` here and then use importlib in _load_metric_conversions?

Reply:
Nice change, it breaks Python3.6 support though. I'm happy to drop Python3.6 support, any thoughts?

Reply:
I would think it makes sense to align to the wider scientific python ecosystem, and given that numpy has dropped support for python 3.6 (see https://numpy.org/neps/nep-0029-deprecation_policy.html), I think it can safely be dropped.

Reply:
Very good point